Dashboard updates (#781)
langchain-ai/langsmith-docs: docs/observability/how_to_guides/dashboards.mdx

- **Prebuilt dashboards**: Automatically generated for every tracing project.
- **Custom dashboards**: Fully configurable collections of charts tailored to your needs.

## Prebuilt dashboards

Prebuilt dashboards are created automatically for each project and cover essential metrics, such as trace count, error rates, token usage, and more.

![prebuilt](./static/prebuilt.gif)

### Dashboard sections

Prebuilt dashboards are broken down into sections that group relevant charts. Navigate between sections using the the navigation at the top.

| Section         | What it shows                                                                                                                                                                                                                                          |
| --------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |

> | Traces          | A [trace](/observability/concepts#traces) is a collection of [runs](/observability/concepts#runs) related to a single operation. For example, if a user request triggers an agent, all runs for that agent invocation would be part of the same trace. |

Reviewer: feel like the 'what is shows' col could describe the actual chart metrics a bit more, eg

Traces: Run counts, error rates, and latency stats for traces. A trace is ...

LLM Calls: Run counts, error rates, and latency stats for LLM runs

Cost & Token: Total and per-trace token counts and costs, broken down by token type

Tools: Run counts, error rates, and latency stats for Tool runs broken down by Tool name. Limited to the 5 most frequently occurring tools

Run Types: Run counts, error rates, and latency stats for runs of any type that occur at depth = 1. Limited to the 5 most frequently occurring run names. Depth refers to how many parents a run has...

Feedback Scores: Aggregate stats for the top 5 most frequently occurring types of feedback. Charts show average score for numerical feedback and category counts for categorical feedback.

| LLM Calls       | All runs where run type is "llm".                                                                                                                                                                                                                      |

> | Cost & Tokens   | Costs are measured using [LangSmith's cost tracking](/observability/how_to_guides/log_llm_trace#manually-provide-token-counts).                                                                                                                        |

Reviewer: not sure this link here is needed / what i would expect to be linked to

> | Tools           | Includes runs where run type is "tool". Limits to top [N] tools.                                                                                                                                                                                       |

Author: @baskaryan Cant remember what the limit is

| Run Types       | Includes runs that are immediate children of the root run. For example, for the following trace, the following runs have a depth of 1: <img src={runDepth} alt="run_depth" style= {{ display: 'block', width: 200 }} />. Limits to top [N] runs.       |
| Feedback Scores | Feedback provided as part of [online evaluators](/observability/how_to_guides/online_evaluations). Limits to top the top 4 feedback scores by volume.                                                                                                  |

### Group by

> Group by [run tag or metadata](/observability/how_to_guides/add_metadata_tags) can be used to split data over arbitrary metrics that are important to your application. The group by setting appears on the top right hand side of the dashboard. Note that the Tool and Run Type charts already have a group by applied, so the global group by won't take effect.

Reviewer: ```suggestion
Group by [run tag or metadata](/observability/how_to_guides/add_metadata_tags) can be used to split data over attributes that are important to your application. The global group by setting appears on the top right hand side of the dashboard. Note that the Tool and Run Type charts already have a group by applied, so the global group by won't take effect; the global group by will apply to all other charts.
```

:::tip

> When adding metadata to runs, we reccomend having the same metadata on the trace, as well has the specific run (eg. LLM call). [WHY]

Author: @baskaryan can you help explain why

Reviewer: Metadata and tags are not propagated from parent to child runs, or vice versa. Therefore if you want to see e.g. both your Trace charts and your LLM Calls charts grouped on some metadata key then both your traces (root runs) and your LLM runs need to have that metadata attached.

:::

> ### Time selector

Reviewer: not specific to prebuilts

Choose a time window in the dropdown on the top right hand side of the dashboard. You can set a default time window, by hovering over the values in the dropdown and clicking "Set default". Maximum time horizon is 30d.

> :::note

Reviewer: feels like this is a part of the Time Selector section, which i assume its not

Reviewer: maybe should have a ### Editing prebuilts (in progress) section that just has this note in it

### Adding charts to your dashboard

1. Within a dashboard, click on the **+ New Chart** button to open up the chart creation pane.
2. Give your chart a name and a description.
3. Configure the chart.

### Chart configuration

#### Select tracing projects and filter runs

- Select one or more tracing projects to track metrics for.
- Use the **Chart filters** section to refine the matching runs. This filter applies to all data series in the chart. For more information on filtering traces, view our guide on [filtering traces in application](./filter_traces_in_application).

#### Pick a metric

- Choose a metric from the dropdown menu to set the y-axis of your chart. With a project and a metric selected, you'll see a preview of your chart and the matching runs.
- For certain metrics (such as latency, token usage, cost), we support comparing multiple metrics with the same unit. For example, you may wany one chart where you can see prompt tokens and completion tokens. Each metric appears as a separate line.

![Multiple metrics](./static/compare_metrics.png)

#### Split the data

There are two ways to create multiple series in a chart (i.e. create multiple lines in a chart):

> 1. **Group by**: Group runs by [run tag or metadata](/observability/how_to_guides/add_metadata_tags), run name or run type. Group by automatically splits the data into multiple series based on the field selected.

Reviewer: ubernit
```suggestion
1. **Group by**: Group runs by [run tag or metadata](/observability/how_to_guides/add_metadata_tags), run name, or run type. Group by automatically splits the data into multiple series based on the field selected.
```

Reviewer: should specify that its alway limited to top 5 of whatever you split on
